spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -20,33 +20,141 @@
+
+[source,java]
+----
+@Incoming("my-channel")                                                // <1>
+public CompletionStage<Void> consume(Message<String> message) {        // <2>
+  return message.ack();
+}
+----
+1. `my-channel` is the channel
+2. the method is called for every message sent to the `my-channel` channel
+
+Reactive Messaging support various form of method signatures.
+This is detailed in the next section.
+
+Remember that Reactive Messaging interactions are made using Reactive Streams.
+A method annotated with `@Incoming` is a Reactive Streams _subscriber_ and so consumes messages according to the request it makes.
+Note that the handling of the Reactive Streams protocol is managed by the Reactive Messaging implementation.
+
+From the user perspective, whether the incoming messages come from a co-located beans or a remote message broker is transparent.
+However, the user may decide to consume a specific subclass of `Message` if the user is aware of this characteristic:
+
+[source,java]
+----
+@Incoming("my-kafka-topic")               
+public CompletionStage<Void> consume(KafkaMessage<String> message) {    // <1>
+  return message.ack();
+}
+----
+1. Explicit consumption of a `KafkaMessage`
 
-Applications declare message subscribers, publishers and processors by annotating methods on CDI beans with `@Incoming` and `@Outgoing` annotations. A method with an `@Incoming` annotation is a message subscriber, it consumes messages from a messaging provider. A method with an `@Outgoing` annotation is a message publisher, it emits messages to a messaging provider. A method with both an `@Incoming` and an `@Outgoing` annotation is a message processor, it consumes messages from one messaging provider, does some transformation to them, and emits messages to another, potentially different, messaging provider.
+==== Message production with @Outgoing
 
-==== Messaging providers
+The `org.eclipse.microprofile.reactive.messaging.Outgoing` annotation is used on a method from a CDI bean to indicate that the method produces messages to a specified channel:
 
-A messaging provider is an abstract concept provided by this spec, it typically will correspond to a messaging transport, such as Apache Kafka, Amazon Kinesis, RabbitMQ or Apache ActiveMQ. It is up an implementation as to which and how many transports it will support. Implementations may provide a default messaging transport, and that default may be configurable, all this is left up to the implementation to decide on the mechanism for providing that. Implementations may also provide an SPI for plugging in custom messaging providers, the definition of that SPI is up to the implementation, and beyond the scope of this specification. A future iteration of this spec may provide such an SPI, for example based on the Java EE Connector Architecture, but currently this is out of scope.
+[source,java]
```

Comment:
(I will need to complete this review (for the changes below this comment) this evening. Looks good so far.)

Reply:
Thanks! I've fixed most of the issue about one which needs a bit more thinking.